sway-core/src/transform/attribute.rs

```diff
@@ -48,7 +48,39 @@ pub enum AttributeKind {
 }
 
 /// Stores the attributes associated with the type.
-pub type AttributesMap = Arc<HashMap<AttributeKind, Vec<Attribute>>>;
+#[derive(Default, Clone, Debug, Eq, PartialEq)]
+pub struct AttributesMap(Arc<HashMap<AttributeKind, Vec<Attribute>>>);
+
+impl AttributesMap {
+    /// Create a new attributes map.
+    pub fn new(attrs_map: Arc<HashMap<AttributeKind, Vec<Attribute>>>) -> AttributesMap {
+        AttributesMap(attrs_map)
+    }
+
+    /// Returns the first attribute by span, or None if there are no attributes.
+    pub fn first(&self) -> Option<(&AttributeKind, &Attribute)> {
+        let mut first: Option<(&AttributeKind, &Attribute)> = None;
```

Comment:
You can avoid imperative mutable state-updating by using `iterator::fold` or `iterator::reduce` in cases like this.

Reply:
Thanks, I'll try this out - not going to block merging this PR on it though :)